File: 1-two-sum.cpp
#include <algorithm>
class Solution {
  public:
    vector<int> twoSum(vector<int>& nums, int target){
        int x{},y{},current{};
        for(int i = 0; i < nums.size() - 1; i++){
            x = i;
            current = target;
            current -=nums[i];
            auto iter =std::find(nums.begin()+i+1,nums.end(), current); 
            if(iter!=nums.end())
            {
                y = std::distance(nums.begin(),iter);
                break;
            }
            
        }
        return vector<int>{x,y};
    }
};

File: 1-two-sum.py
# O(n^2)
class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        for index, val in enumerate(nums):
            tmp = target - val
            try:
                return [index, nums.index(tmp, index + 1)]
            except ValueError:
                continue


# O(n)
class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        vals = {}
        for index, val in enumerate(nums):
            if target - val in vals:
                return [index, vals[target - val]]
            else:
                vals[val] = index
